### PR TITLE
58953 - Keep the tax move line description when we generate the contr…

### PR DESCRIPTION
**Pull request by paa-axelor**

### CODE REVIEW

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineTaxServiceImpl.java [hunk 1]
```diff
@@ -253,6 +259,14 @@ public int compare(MoveLine o1, MoveLine o2) {
       }
     }
 
+    if ("".equals(oldTaxDescription)) {
```
Comment (jmi-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineTaxServiceImpl.java:262]: You think it's better to use Strings.isNullOrEmpty(oldTaxDescription)

**Review by jmi-axelor — CHANGES_REQUESTED** [review]

File: axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveToolServiceImpl.java [hunk 2]
```diff
@@ -510,9 +507,18 @@ public void setOriginOnMoveLineList(Move move) {
   }
 
   @Override
+  @Transactional
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveToolServiceImpl.java]: Removed

File: axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineTaxServiceImpl.java (hunk 1 above)
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineTaxServiceImpl.java]: Fixed

File: axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveToolServiceImpl.java [hunk 3]
```diff
@@ -510,9 +507,18 @@ public void setOriginOnMoveLineList(Move move) {
...
   public void setDescriptionOnMoveLineList(Move move) {
+    Move defaultMove = null;
+    if (move.getId() != null) {
+      defaultMove = Beans.get(MoveRepository.class).find(move.getId());
```
[the remark this answers is not in the document]
Comment (paa-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveToolServiceImpl.java]: Fixed

**Review by jmi-axelor — APPROVED** [review]

**Pull request closed without merging**